File: ops/backend/app/schemas/machine.py
"""Machine (host) schemas."""

from pydantic import BaseModel

from app.schemas.common import Status


class Machine(BaseModel):
    """A monitored host: VPS, cloud instance or workstation."""

    id: str
    name: str
    location: str
    provider: str
    status: Status
    cpu: float | None = None
    ram: float | None = None
    disk: float | None = None
    temperatureC: float | None
    internetMs: float | None = None
    brokerPingMs: float | None = None
    pythonStatus: Status
    uptimeSec: int | None = None
    lastHeartbeat: str | None = None
    strategyCount: int | None = None
    agentId: str | None = None
    agentVersion: str | None = None
    hostname: str | None = None
    environment: str | None = None
    lastEvent: str | None = None
    lastTrade: str | None = None
    lastError: str | None = None
    lastSuccessfulUpload: str | None = None
    queueDepth: int | None = None
    oldestPendingAgeSec: int | None = None
    transportState: str | None = None
    currentSessionId: str | None = None
    tradingProcessState: str | None = None
    lastEodSync: str | None = None
    lastEodStatus: str | None = None
    recoveryState: str | None = None
    lastRecovery: str | None = None
    eventsRecovered: int | None = None
    eodBacklog: int | None = None

File: ops/frontend/src/types/machine.ts
import type { Status } from './common'

/** A monitored host (VPS, cloud instance or local workstation). */
export interface Machine {
  id: string
  name: string
  /** Geographic / provider label, e.g. `London · Equinix LD4`. */
  location: string
  provider: string
  status: Status
  /** Percentages (0–100). */
  cpu: number
  ram: number
  disk: number
  /** CPU temperature in °C (null when the host does not report it). */
  temperatureC: number | null
  /** Round-trip to the data feed in ms. */
  internetMs: number
  /** Round-trip to the broker gateway in ms. */
  brokerPingMs: number
  /** Whether the Python trading runtime is alive. */
  pythonStatus: Status
  /** Uptime in seconds. */
  uptimeSec: number
  /** ISO timestamp of the last heartbeat. */
  lastHeartbeat: string
  /** Number of strategies hosted on this machine. */
  strategyCount: number
  agentId?: string | null
  agentVersion?: string | null
  hostname?: string | null
  environment?: string | null
  lastEvent?: string | null
  lastTrade?: string | null
  lastError?: string | null
  lastSuccessfulUpload?: string | null
  queueDepth?: number | null
  oldestPendingAgeSec?: number | null
  transportState?: string | null
  currentSessionId?: string | null
  tradingProcessState?: string | null
  lastEodSync?: string | null
  lastEodStatus?: string | null
  recoveryState?: string | null
  lastRecovery?: string | null
  eventsRecovered?: number | null
  eodBacklog?: number | null
}
